streamr-dev/streamr-platform
# [PLATFORM-33] Start/stop canvases + start options

* Enables Start/Stop buttons
* Enables historical "speed" settings
* Enables/disables "reset & start" functionality based on canvas "serialised" status.
* Disables editing & undo while canvas is running.
* Adds temporary swapping to historical canvases.
* Adds workaround for serverside `/canvases?adhoc=false` responding with `adhoc` canvases.
* Adds tests around undo/redo functionality.
* Automatically subscribes to streams for running canvas
* Automatically unsubscribes if running canvas ends.
* Automatically reverts to "parent" canvas if a historical canvas is ended.

To Test:

1. Open Any Canvas.
2. Create a new Canvas via the "…" menu  on the left of the canvas.
2. Add a clock & a table module.
2. Change the clock's unit to "hour" ("second" will flood the dev env with too many events and will require restarting build 👎).
2. Connect the modules via date or timestamp.
2. Switch to Realtime mode.
2. Click Start, see it's running. Editing & undo should be disabled. 
2. Click Stop. Editing & undo should be restored.
2. Switch to Historical mode.
2. Fill the `From` and `To` dates with a date formatted like: `2018-11-20`. The historical mode `From` & `To` fields need a date picker & default value. TODO. 
2. Click Start. Canvas should start then almost immediately end on its own. This is expected behaviour.

No data will display in the table module as I have no hooked up any custom modules like that yet. This is PR is just getting basic start/stop + subscriptions hooked up.
